File: spec/functional_test/fixtures/java/spark/src/Application.java
package com.example;

import static spark.Spark.*;

public class Application {
    private static final String API_PREFIX = "/api";
    private static final String REPORTS_PATH = "/reports";
    private static final String SOCKET_PATH = "/events";
    private static final String TRACE_HEADER = "X-Report-Trace";

    static final class RouteParts {
        static final String DETAIL = "/:reportId";
    }

    public static void main(String[] args) {
        staticFiles.location("/public");
        staticFiles.externalLocation("/var/www/public");

        get("/hello", (req, res) -> {
            String name = req.queryParams("name");
            return "Hello " + name;
        });

        post("/users", (req, res) -> {
            String body = req.body();
            res.status(201);
            return body;
        });

        path("/api", () -> {
            get("/status", (req, res) -> "ok");
            path("/v1", () -> {
                get("/health", (req, res) -> "healthy");
                post("/submit", (req, res) -> {
                    String body = req.body();
                    String token = req.headers("X-Token");
                    return body;
                });
                get("/items/:itemId", (req, res) -> {
                    String category = req.queryParams("category");
                    return "ok";
                });
            });
        });

        delete("/sessions/:id", (req, res) -> {
            String session = req.cookie("session");
            res.status(204);
            return "";
        });

        Spark.put("/profile", (req, res) -> {
            String body = req.body();
            String trace = req.headers("X-Trace");
            return body;
        });

        redirect.get("/legacy-home", "/hello");
        redirect.post("/legacy-submit", "/api/v1/submit", Redirect.Status.SEE_OTHER);
        redirect.any("/legacy-any", "/hello", Redirect.Status.MOVED_PERMANENTLY);

        path(API_PREFIX, () -> {
            get(REPORTS_PATH + RouteParts.DETAIL, (req, res) -> {
                String trace = req.headers(TRACE_HEADER);
                return trace;
            });
            webSocket(SOCKET_PATH + "/:roomId", EventsSocket.class);
        });
    }

    static class EventsSocket {
    }
}

File: spec/functional_test/fixtures/kotlin/http4k/src/main/kotlin/com/example/Application.kt
package com.example

import org.http4k.core.Method.DELETE
import org.http4k.core.Method.GET
import org.http4k.core.Method.PATCH
import org.http4k.core.Method.POST
import org.http4k.core.Method.PUT
import org.http4k.core.Request
import org.http4k.core.Response
import org.http4k.core.Status.Companion.OK
import org.http4k.routing.bind
import org.http4k.routing.routes

val app = routes(
    "/hello" bind GET to { req: Request ->
        val name = req.query("name")
        Response(OK).body("hello $name")
    },

    "/users" bind POST to { req: Request ->
        val body = req.bodyString()
        Response(OK).body(body)
    },

    "/users/{id}" bind PUT to { req: Request ->
        val token = req.header("X-API-Key")
        val body = req.bodyString()
        Response(OK)
    },

    "/api" bind routes(
        "/status" bind GET to { _: Request -> Response(OK) },
        "/v1" bind routes(
            "/health" bind GET to { _: Request -> Response(OK) },
            "/submit" bind POST to { req: Request ->
                val body = req.bodyString()
                val token = req.header("X-Token")
                Response(OK)
            },
            "/items/{itemId}" bind GET to { req: Request ->
                val category = req.query("category")
                Response(OK)
            }
        )
    ),

    "/sessions/{id}" bind DELETE to { req: Request ->
        Response(OK)
    },

    "/profile" bind PATCH to { req: Request ->
        val email = req.form("email")
        val phone = req.form("phone")
        Response(OK)
    }
)
